src/compiler/core.ts

```diff
@@ -2985,7 +2985,8 @@ namespace ts {
      */
     export function matchedText(pattern: Pattern, candidate: string): string {
         Debug.assert(isPatternMatch(pattern, candidate));
-        return candidate.substr(pattern.prefix.length, candidate.length - pattern.suffix.length);
+        const matchLength = candidate.length - pattern.suffix.length - pattern.prefix.length;
+        return candidate.substr(pattern.prefix.length, matchLength);

```

Comment:
Could use `slice` which takes start and end and is more common in the rest of the codebase. `substr` leads to nothing but problems (#20578).

Reply:
I think you meant `substring` not slice `slice` but i get your point :)

Reply:
Looks like we use them both about as often. According to [this](https://stackoverflow.com/questions/2243824/what-is-the-difference-between-string-slice-and-string-substring#2243835) `slice` and `substring` are basically equivalent with the exception of error handling -- if you avoid negative numbers, `NaN`, and `start > end`, they're the same.